src/dom/document.ts

```diff
@@ -307,8 +307,25 @@ export class Document extends Node {
   private _getElementsByClassName(className: string, search: Node[]): Node[] {
     for (const child of this.childNodes) {
       if (child.nodeType === NodeType.ELEMENT_NODE) {
-        if ((<Element> child).classList.contains(className)) {
-          search.push(child);
+        // remove duplicate class names
+        const classList = new Set(className.split(" "));
+
+        let matches: Node[] = [];
+        let matchesCount = 0;
+
+        for (const singleClassName of classList) {
+          if ((<Element> child).classList.contains(singleClassName)) {
+            matchesCount++;
+            // avoid pushing the same element multiple times due to multiple class names
+            if (!matches.includes(child)) {
+              matches.push(child);
```

Comment:
You don't need `matches` here after removing the `Set` above, since if `matchesCount` === `classList.length` (as an array, and not a `Set`) then the element must match. 

So you can remove `let matches: Node[] = []` and this `if` statement (after making `classList` a regular array without the `Set`)